### PR TITLE
Added Brian Kernighan Algorithm

### PR DESCRIPTION
**Pull request by debanjan-2002**
Sir,
Upon your change requests, I have made the changes and made a new pull request.
Things added - 
1) Brian's Kernighan's Algorithm
2) Updated readme.

Fixes - #44 

I have written a brief code following all the code of conduct.
 Regards,
Debanjan Poddar

### CODE REVIEW

**Comment by debanjan-2002** [comment]
Sir,
I have made the necessary changes that you mentioned. Please review this pull request.
Regards,
Debanjan Poddar
Fixes - #44

File: Algorithm/Bit_manipulation/readme.md [hunk 1]
```diff
@@ -52,3 +52,5 @@ There are different bitwise operations used in the bit manipulation. Bit operati
 
 
 ## Questions :
+
```
Comment (raksha009) [inline Algorithm/Bit_manipulation/readme.md:55]: Remove extra line

File: Code/C++/Brian_Kernighan.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+    Below is the code (C++) representing the idea.
+    There is a function count_set(), that takes an integer as argument and
+    returns the number of set bits.
+*/
+#include<bits/stdc++.h>
+using namespace std;
+
+int count_set(int n){
+    int ans=0;      // This keeps the track of the number of set bits
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:39]: Add comment above the line

File: Code/C++/Brian_Kernighan.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    There is a function count_set(), that takes an integer as argument and
+    returns the number of set bits.
+*/
+#include<bits/stdc++.h>
+using namespace std;
+
+int count_set(int n){
+    int ans=0;      // This keeps the track of the number of set bits 
+    while(n>0){
+        n=(n&(n-1));
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:41]: give spaces around the operator.

File: Code/C++/Brian_Kernighan.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+/* TEST CASES
+
+    1) input   : 32
+       output  : 1
+    2) input   : 39
+       output  : 4
+
+
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:62]: Remove extra spacing

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Algorithm/Bit_manipulation/readme.md (hunk 1 above)
Comment (debanjan-2002) [reply Algorithm/Bit_manipulation/readme.md]: @raksha009 
I have made all the necessary changes that you have told about.
Please review the changes and let me know if any further changes are required or not.
I am new to this and I am having a good time learning about these things.
Regards,
Debanjan Poddar
Fixes - #44

File: Code/C++/Brian_Kernighan.cpp (hunk 2 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: @raksha009 
I have made all the necessary changes that you have told about.
Please review the changes and let me know if any further changes are required or not.
I am new to this and I am having a good time learning about these things.
Regards,
Debanjan Poddar
Fixes - #44

File: Code/C++/Brian_Kernighan.cpp (hunk 4 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: @raksha009 
I have made all the necessary changes that you have told about.
Please review the changes and let me know if any further changes are required or not.
I am new to this and I am having a good time learning about these things.
Regards,
Debanjan Poddar
Fixes - #44

File: Code/C++/Brian_Kernighan.cpp (hunk 3 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: @raksha009 
I have made all the necessary changes that you have told about.
Please review the changes and let me know if any further changes are required or not.
I am new to this and I am having a good time learning about these things.
Regards,
Debanjan Poddar
Fixes - #44

File: Code/C++/Brian_Kernighan.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
+// Brian Kernighan's Algorithm
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:1]: Include this in multiline comment itself.

File: Code/C++/Brian_Kernighan.cpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+/* TEST CASES
+    1) input   : 32
+       output  : 1
+    2) input   : 39
+       output  : 4
+
+        TIME COMPLEXITY - O(No. of set bits in input)
+        SPACE COMPLEXITY - O(1)
+*/
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:65]: Add a new line here

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/C++/Brian_Kernighan.cpp (hunk 5 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: I have made the necessary changes as requested. Please let me know if any further changes are required or not.
Regards,
Debanjan Poddar
@raksha009 
@tarun26091999

File: Code/C++/Brian_Kernighan.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+             n&(n-1) ->   000...100000
+             
+             We clearly see the only change tha happens after the 2 steps,
+             is that the left most set bit gets turned off.
+        
+        3) We repeat this step until n becomes 0.
+
+    
+    Below is the code (C++) representing the idea.
+    There is a function count_set(), that takes an integer as argument and
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:32]: Plz remove line 31.
Add comment on line 32 above func at line 38

File: Code/C++/Brian_Kernighan.cpp [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    return ans;
+}
+
+//main function
+int main(){
+    int n;
+    // Taking input from the user
+    cin>>n;   
+    int ans = count_set(n);
+    cout<<ans<<"\n";
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:54]: Add spaces around the operators like << , =, > etc.

File: Code/C++/Brian_Kernighan.cpp (hunk 6 above)
Comment (raksha009) [reply Code/C++/Brian_Kernighan.cpp:67]: Add a new line after line 67.

File: Code/C++/Brian_Kernighan.cpp [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+/* TEST CASES
+    1) input   : 32
+       output  : 1
+    2) input   : 39
+       output  : 4
+
+        TIME COMPLEXITY -> O(No. of set bits in input)
```
Comment (raksha009) [inline Code/C++/Brian_Kernighan.cpp:63]: Write it as TIME COMPLEXITY -> O(n) where n is No. of set bits in input

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/C++/Brian_Kernighan.cpp (hunk 7 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: Made the necessary changes. Please review it and if more changes are required then please kindly inform me.
Regards,
Debanjan Poddar.
@raksha009 
@tarun26091999

File: Code/C++/Brian_Kernighan.cpp (hunk 8 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: Made the necessary changes. Please review it and if more changes are required then please kindly inform me.
Regards,
Debanjan Poddar.
@raksha009 
@tarun26091999

File: Code/C++/Brian_Kernighan.cpp (hunk 9 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: Made the necessary changes. Please review it and if more changes are required then please kindly inform me.
Regards,
Debanjan Poddar.
@raksha009 
@tarun26091999

**Review by raksha009 — APPROVED** [review]

**Comment by debanjan-2002** [comment]
**Please merge this pull request, as you have approved it**. It would be really nice if you do so. 
Regards,
Debanjan Poddar
@raksha009 
@tarun26091999

File: Code/C++/Brian_Kernighan.cpp [hunk 10]
```diff
@@ -49,9 +47,9 @@ int count_set(int n){
...
     int n;
     // Taking input from the user
-    cin>>n;   
+    cin >> n;   
     int ans = count_set(n);
-    cout<<ans<<"\n";
+    cout << ans << "\n";
 }
 
 /* TEST CASES
```
Comment (rudrakshi99) [inline Code/C++/Brian_Kernighan.cpp:55]: use lowercase as `Test case :`

File: Code/C++/Brian_Kernighan.cpp [hunk 11]
```diff
@@ -60,8 +58,8 @@ int main(){
     2) input   : 39
        output  : 4
 
-        TIME COMPLEXITY -> O(No. of set bits in input)
+        TIME COMPLEXITY -> O(n) where n is No. of set bits in input.
         
         SPACE COMPLEXITY -> O(1)
```
Comment (rudrakshi99) [inline Code/C++/Brian_Kernighan.cpp:63]: lowercase!

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

File: Code/C++/Brian_Kernighan.cpp (hunk 10 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: I have made the changes that were asked. Please review it and let me know if any other changes are required. I am open to learning. If it is okay I would request you to merge this.  
Regards,
Debanjan Poddar
@rudrakshi99 
@tarun26091999 
@plazzy99 
@raksha009

File: Code/C++/Brian_Kernighan.cpp (hunk 11 above)
Comment (debanjan-2002) [reply Code/C++/Brian_Kernighan.cpp]: I have made the changes that were asked. Please review it and let me know if any other changes are required. I am open to learning. If it is okay I would request you to merge this.  
Regards,
Debanjan Poddar
@rudrakshi99 
@tarun26091999 
@plazzy99 
@raksha009

**Review by rudrakshi99 — APPROVED** [review]

**Pull request merged**